src/modules/julia.rs

```diff
@@ -0,0 +1,558 @@
+
+        update_project_and_manifest(project).await;
+
+        csleep(rng.gen_range(10..100)).await;
+
+        report_packages(packages).await;
+
+        csleep(rng.gen_range(150..500)).await;
+
+        build_artifacts(artifacts).await;
+
+        csleep(rng.gen_range(150..500)).await;
+
+        precompile(packages).await;
+
+        if rng.gen::<f32>() < 0.25f32 {
+            gc().await;
+        }
+
+        csleep(rng.gen_range(50..250)).await;
+
+        print_pkg_prompt(project, false).await;
+
+        // wait user input
+        csleep(rng.gen_range(500..5000)).await;
+
+        // exit pkg mode
+        print_julia_prompt(true).await;
+
+        // wait user input
+        csleep(rng.gen_range(1000..7000)).await;
+
+        // quit julia
+        newline().await;
+
+        if appconfig.should_exit() {
+            return;
+        }
+    }
+}
+
+async fn print_banner() {
+    print(format!(
+        r#"               {gu}
+   {bu}       _ {ru}{g}{mu}     |  Documentation: https://docs.julialang.org
+  {b}     | {r} {m}    |
+   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
+  | | | | | | |/ _` |  |
+  | | |_| | | | (_| |  |  Version 1.7.3 (2022-05-06)
+ _/ |\__'_|_|_|\__'_|  |  Fedora 35 build
+|__/                   |
+"#,
+        r = Paint::red("(_)").bold(),
+        ru = Paint::red("_").bold(),
+        g = Paint::green("(_)").bold(),
+        gu = Paint::green("_").bold(),
+        b = Paint::blue("(_)").bold(),
+        bu = Paint::blue("_").bold(),
+        m = Paint::magenta("(_)").bold(),
+        mu = Paint::magenta("_").bold(),
+    ))
+    .await;
+}
+
+async fn print_julia_prompt(overwrite: bool) {
+    if overwrite {
+        erase_line().await;
+    } else {
+        newline().await;
+    }
+    print(format!("{} ", Paint::green("julia>").bold())).await;
+}
+
+async fn print_pkg_prompt(project: &str, overwrite: bool) {
+    if overwrite {
+        erase_line().await;
+    } else {
+        newline().await;
+    }
+    print(format!(
+        "{} ",
+        Paint::blue(format!("({}) pkg>", project)).bold()
+    ))
+    .await;
+}
+
+async fn log_action(action: impl Display, message: impl Display) {
+    print(format!("{:>12} {}", Paint::green(action).bold(), message)).await;
+    newline().await;
+}
+
+async fn log_progress(bar: &progress_string::Bar) {
+    print(format!("{:>10} {}", Paint::cyan("Progress").bold(), bar)).await;
+    newline().await;
+}
+
+async fn install_packages(packages: &[&Package<'_>]) {
+    let mut rng = thread_rng();
+
+    let max_name_length = packages.iter().map(|p| p.name.len()).max().unwrap();
+    for package in packages {
+        let package_and_version = format!(
+            "{name} {empty:─>width$} v{version}",
+            name = package.name,
+            width = max_name_length - package.name.len() + 1,
+            empty = "",
+            version = package.versions.last().unwrap()
+        );
+
+        if rng.gen::<f32>() < 0.1f32 {
```

Comment:
I think it's more idiomatic to do
```suggestion
        if rng.gen_bool(1.0 / 10.0) {
```
See [here](https://docs.rs/rand/latest/rand/trait.Rng.html#method.gen_bool).

You've done this in a few other places as well so please check those, too.

Reply:
Thanks, I wasn't aware of it.

Initially I used [rand::distributions::Bernoulli::new](https://docs.rs/rand/latest/rand/distributions/struct.Bernoulli.html#method.new) everywhere, but then got bothered by having to `unwrap` the result.